main.go

```diff
@@ -21,43 +21,32 @@ type Host struct {
 	NoAuth  bool     `json:"noAuth"`
 }
 
+type User struct {
+	Type      string `json:"type"`
+	PublicKey string `json:"publicKey"`
+	Name      string `json:"name"`
+}
+
 var configFile = flag.String("config", "", "User-supplied configuration file to use")
 
-func parseAuthFile(filename string) ([]*sshmux.User, error) {
+func parseUsers() ([]*sshmux.User, error) {
 	var users []*sshmux.User
-
-	authFile, err := ioutil.ReadFile(filename)
+	us := make([]User, 0)
+	err := viper.UnmarshalKey("users", &us)
 	if err != nil {
 		return nil, err
 	}
-
-	// Parse authfile as authorized_key
-
-	for len(authFile) > 0 {
-		switch authFile[0] {
-		case '\n', '\r', '\t', ' ':
-			authFile = authFile[1:]
-			continue
-		}
-
-		var (
-			pk      ssh.PublicKey
-			comment string
-		)
-
-		pk, comment, _, authFile, err = ssh.ParseAuthorizedKey(authFile)
+	for _, u := range us {
+		pk, _, _, _, err := ssh.ParseAuthorizedKey([]byte(u.Type + " " + u.PublicKey))
```

Comment:
I think you can just use [`ssh.ParsePublicKey`](https://godoc.org/golang.org/x/crypto/ssh#ParsePublicKey) here. [`ssh.ParseAuthorizedKeys`](https://godoc.org/golang.org/x/crypto/ssh#ParseAuthorizedKey) just extract option, comments and type, and give the remainder to [`ssh.ParsePublicKey`](https://godoc.org/golang.org/x/crypto/ssh#ParsePublicKey)... If I read the code correctly, that is. The parsing logic in there is not super pretty.